File: 74-SearchA2dMatrix/74-SearchA2dMatrix.java
// Last updated: 7/21/2026, 5:01:59 PM
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        
        int rows = matrix.length;
        int cols = matrix[0].length;

        for(int i = 0; i < rows * cols; i++){
            if(matrix[i / cols][i % cols] == target){
                return true;
            }
        }

        return false;
    }
}

File: 74-SearchA2dMatrix/74-SearchA2dMatrix.cpp
// Last updated: 7/21/2026, 5:02:02 PM
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        bool ispresent = false;
        for (int i = 0; i < matrix.size(); i++) {
            for (int j = 0; j < matrix[i].size(); j++) {     
                if(matrix[i][j] == target){
                    ispresent = true;
                    return ispresent;
                }
            }
        }
        return ispresent;
    }
};